### PR TITLE
Platform 293

### PR DESCRIPTION
**Pull request by jamccomb92**
### Description of proposed changes
_What does this pull request accomplish? Why should it be integrated?_
Creates initial settings view for users.

### Checklist
_Put an `x` in the boxes that apply._
- [x] My lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my changes work (if applicable)
- [ ] I have added or updated necessary documentation (if appropriate)

### Additional comments
_Is there anything else you want us to know?_
I do believe the UI could and should be cleaned up. Went for function over style towards the end. Looks decent but could be upgraded in places

### CODE REVIEW

File: domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  Button, Headline, IconButton, Text, TextInput
+} from 'react-native-paper';
+
+import { getData, storeData } from '../../../../../modules/async-storage';
+import I18n from '../../../../../modules/i18n';
+import { theme } from '../../../../../modules/theme';
+
+const FindRecords = () => {
+  useEffect(() => {
+    async function setUserInformation() {
```
Comment (hopetambala) [inline domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js:15]: did you check that running an async function in the useEffect doesn't introduce a memory leak?

File: domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    marginTop: 10,
+    marginBottom: 10,
+  },
+  horizontalLinePrimary: {
+    borderBottomColor: theme.colors.primary,
+    borderBottomWidth: 1,
+    marginTop: 10,
+    marginBottom: 10,
+  }
+});
```
Comment (hopetambala) [inline domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js:208]: let's move this to an `index.style.js` file

File: domains/Settings/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            logOut={logOut}
+            settingsView={settingsView}
+            setSettingsView={setSettingsView}
+          />
+        )}
+        {settingsView === 'Support' && (
+          <View>
+            <View style={{ flexDirection: 'row', marginLeft: 'auto', marginRight: 'auto' }}>
+              <View style={{ paddingRight: '5%' }}>
+                <Button onPress={() => setSettingsView('Settings')}>Settings</Button>
```
Comment (hopetambala) [inline domains/Settings/index.js:29]: Spanish Translation

File: domains/Settings/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+              </View>
+              <View style={{ paddingLeft: '5%' }}>
+                <Button mode="contained">Support</Button>
```
Comment (hopetambala) [inline domains/Settings/index.js:32]: Spanish Translation

File: domains/Settings/index.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+              </View>
+            </View>
+            <Button onPress={() => {
+              setView(prevView);
+            }}
+            >
+              Back
+            </Button>
+            <Button onPress={logOut}>Logout</Button>
```
Comment (hopetambala) [inline domains/Settings/index.js:41]: Spanish Translation

File: domains/Settings/SettingsHome/index.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    color: '#555',
+    marginVertical: 7,
+  },
+  horizontalLineGray: {
+    borderBottomColor: '#D0D0D0',
+    borderBottomWidth: 1,
+    marginTop: 10,
+    marginBottom: 10,
+  },
+});
```
Comment (hopetambala) [inline domains/Settings/SettingsHome/index.js:142]: let's move this as well to a `index.style.js`

**Review by hopetambala — CHANGES_REQUESTED** [review]
Overall I like the architecture! I'm going to do some more digging tonight and tomorrow before we release with the others. There's alot of code!

File: domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js (hunk 1 above)
Comment (jamccomb92) [reply domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js:15]: yes, that was the proper way to perform the async within a useEffect from what i found online (here's example): https://javascript.plainenglish.io/how-to-use-async-function-in-react-hook-useeffect-typescript-js-6204a788a435

File: domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js (hunk 2 above)
Comment (jamccomb92) [reply domains/Settings/SettingsHome/AccountSettings/FindRecords/index.js:208]: done

File: domains/Settings/index.js (hunk 3 above)
Comment (jamccomb92) [reply domains/Settings/index.js:29]: this isnt part of the PR, hid the support view for now

File: domains/Settings/index.js (hunk 4 above)
Comment (jamccomb92) [reply domains/Settings/index.js:32]: this isnt part of the PR, hid the support view for now

File: domains/Settings/index.js (hunk 5 above)
Comment (jamccomb92) [reply domains/Settings/index.js:41]: this isnt part of the PR, hid the support view for now

File: domains/Settings/SettingsHome/index.js (hunk 6 above)
Comment (jamccomb92) [reply domains/Settings/SettingsHome/index.js:142]: moved them all to same index.style.js

File: domains/DataCollection/index.js [hunk 7]
```diff
@@ -35,6 +36,7 @@ const puenteForms = [
 const DataCollection = ({ navigation }) => {
   const [scrollViewScroll, setScrollViewScroll] = useState();
   const [view, setView] = useState('Root');
+  const [prevView, setPrevView] = useState('Root');
```
Comment (hopetambala) [inline domains/DataCollection/index.js:39]: @jamccomb92 why do we need a separate state object for prevView?

File: domains/Settings/SupportHome/SupportSettings/Feedback/index.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import {
+  View
+} from 'react-native';
+import email from 'react-native-email';
```
Comment (hopetambala) [inline domains/Settings/SupportHome/SupportSettings/Feedback/index.js:5]: hmm this library isn't the most supported, I think you mentioned that you were trying to use [expo mail composer right?](https://docs.expo.io/versions/latest/sdk/mail-composer/). Why didn't you use that again?

**Review by hopetambala — CHANGES_REQUESTED** [review]
Just a little question or so about the react-native-email library, but the code seems straightforward! Testing the UI now

**Review by hopetambala — APPROVED** [review]
UI is beautiful! I think this will work great, I'm going to test on my other phones and then if it works we can merge!

I got two errors related to the new build configuration that I have to fix, but then this should be good to go

File: domains/Settings/SupportHome/index.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+                <View style={styles.horizontalLineGray} />
+              </View>
+            ))}
+          </View>
+          <Button onPress={() => {
+            setView(prevView);
+          }}
+          >
+            {I18n.t('accountSettings.back')}
+          </Button>
```
Comment (hopetambala) [inline domains/Settings/SupportHome/index.js:101]: What's the point of `prevView`?

I think you can treat the settings button more like a toggle button (each press either pops up the settings page or closes it). 

Or if you want to keep your current approach, the back buttons need to be working (they didn't work for me) when I wanted to go back to the main data collection page

**Review by hopetambala — CHANGES_REQUESTED** [review]

File: domains/Settings/index.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import {
+  View
+} from 'react-native';
+
+import SettingsHome from './SettingsHome';
+import SupportHome from './SupportHome';
+
+const SettingsView = ({
+  setView, prevView, logOut, setSettings
```
Comment (hopetambala) [inline domains/Settings/index.js:10]: @jamccomb92 make sure to remove unused passed props! Linting won't catch these

File: domains/Settings/index.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+}) => {
+  const [settingsView, setSettingsView] = useState('Settings');
+  return (
+    <View>
+      <View style={{ paddingTop: '7%' }}>
+        {settingsView === 'Settings' && (
+          <SettingsHome
+            setView={setView}
+            prevView={prevView}
```
Comment (hopetambala) [inline domains/Settings/index.js:19]: @jamccomb92 here too

File: domains/Settings/index.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+            setSettingsView={setSettingsView}
+            setSettings={setSettings}
+          />
+        )}
+        {settingsView === 'Support' && (
+          <SupportHome
+            settingsView={settingsView}
+            setSettingsView={setSettingsView}
+            setView={setView}
+            prevView={prevView}
```
Comment (hopetambala) [inline domains/Settings/index.js:31]: @jamccomb92 here too

**Review by hopetambala — CHANGES_REQUESTED** [review]
quick changes and we're good!

File: domains/Settings/index.js (hunk 12 above)
Comment (jamccomb92) [reply domains/Settings/index.js]: Good Catch.

**Review by hopetambala — APPROVED** [review]

**Pull request merged**